controller.go

```diff
@@ -181,6 +260,60 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 	return network, nil
 }
 
+func (c *controller) newNetworkFromStore(n *network) {
+	c.Lock()
+	defer c.Unlock()
+
+	if _, ok := c.drivers[n.networkType]; !ok {
+		log.Warnf("Network driver unavailable for type=%s. ignoring network updates for %s", n.Type(), n.Name())
+		return
+	}
+	n.ctrlr = c
+	n.driver = c.drivers[n.networkType]
+	c.networks[n.id] = n
+	// TODO : Populate n.endpoints back from endpoint dbstore
+}
+
+func (c *controller) addNetworkToStore(n *network) error {
+	if isReservedNetwork(n.Name()) {
+		return nil
+	}
+	c.Lock()
+	cs := c.store
+	c.Unlock()
+	if cs == nil {
+		log.Debugf("datastore not initialized. Network %s is not added to the store", n.Name())
+		return nil
+	}
+	return cs.PutObjectAtomic(n)
+}
+
+func (c *controller) watchNewNetworks() {
+	c.Lock()
+	cs := c.store
+	c.Unlock()
+
+	cs.KVStore().WatchRange(datastore.Key("network"), "", 0, func(kvi []store.KVEntry) {
+		for _, kve := range kvi {
+			var n network
+			err := json.Unmarshal(kve.Value(), &n)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+			n.dbIndex = kve.LastIndex()
+			c.Lock()
+			existing, ok := c.networks[n.id]
+			c.Unlock()
+			if ok && existing.dbIndex == n.dbIndex {
```

Comment:
Okay. But, currently we are controlling the store update on network just for creation. Since there are no updates, this event should not happen. But, I do get your point. I will handle that case and log a debug message. (We might end up in this situation for partition recovery cases... but we have to handle that regardless with the discussed `status` and `alarm` mechanisms. Does that sound right to you ?

The reason I didnt add the `delete` yet is the interactions that it need with the driver code and also it will have some impact on presence of `endpoint`.

If we have to handle `delete` in this PR, then we have to deal with other cases as well such as `driver integration` and `endpoints` handling. If you are okay, we can handle it right after this PR. If not, I will have to spend more time on it.